src/satellite_images_nso/api/nso_georegion.py

```diff
@@ -334,6 +400,31 @@ def execute_link(
             else:
                 cropped_path = nso_manipulator.add_height(cropped_path, add_height_band)
 
+        if fill_with_nearest_date:
+            print(
+                "-----Filling satellite image with data from the nearest other satellite----"
```

Comment:
Perhaps I am missing something, but there is 0 guarantee that the data from the nearest other satellite contains a different part of the same georegion, right? It could be the same part of it or even less. So why do we want to blindly assume this improves the amount of data? I think it requires another step to check for that.

Reply:
it should raise a error that no links are found which contain the missing region in the satellite image. But indeed there is no guarantee for it.

Reply:
also the api of NSO return the polygon on the satellite image and thus we can verify that it has the missing region